File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpPlacePaletteRequest.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;
import java.util.Map;

public record RMcpPlacePaletteRequest(
        Map<String, Entry> palette,
        List<Target> targets,
        boolean dryRun
) {
    public record Entry(
            String blockId,
            Map<String, String> state
    ) {
    }

    public record Target(
            RMcpBlockPosData pos,
            String key
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpPlacePaletteRequest.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpPlacePaletteRequest(
    val palette: Map<String, Entry>,
    val targets: List<Target>,
    val dryRun: Boolean
) {
    @JvmRecord
    data class Entry(
        val blockId: String,
        val state: Map<String, String>?
    )

    @JvmRecord
    data class Target(
        val pos: RBlockPos,
        val key: String?
    )
}
